pkg/client/client.go

```diff
@@ -56,43 +57,59 @@ func newUnixSocketClient(ctx context.Context, config *Config) (*Client, error) {
 	}, nil
 }
 
-func newNetworkClient(ctx context.Context, config *Config) (*Client, error) {
+func newNetworkClient(ctx context.Context, config *Config) (*Client, error) {  
+
 	certificate, err := tls.LoadX509KeyPair(
 		config.CertFile,
 		config.KeyFile,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("error loading the X.509 key pair: %v", err)
 	}
+	if(config.InsecureSkipMutualTLSAuth){
+		transportCreds := credentials.NewTLS(&tls.Config{
+			ServerName:   config.Hostname,
+			Certificates: []tls.Certificate{certificate},
+			InsecureSkipVerify: true, 
+
+		})
+		dialOptions := append(config.DialOptions, grpc.WithTransportCredentials(transportCreds))
+		conn, err := grpc.DialContext(ctx, fmt.Sprintf(targetFormat, config.Hostname, config.Port), dialOptions...) 
+		if err != nil {
+			return nil, fmt.Errorf("error dialing server: %v", err)
+		}
+		return &Client{
+			conn: conn,
+		}, nil
+	}
```

Comment:
While a working solution, this change can be simplified using something along the lines of:
```
        tlsConfig := tls.Config {
		ServerName:   config.Hostname,
		Certificates: []tls.Certificate{certificate},
	}

	if(config.InsecureSkipMutualTLSAuth){
		tlsConfig.InsecureSkipVerify = true
	} else {
		certPool := x509.NewCertPool()
		rootCA, err := ioutil.ReadFile(config.CARootFile)
		if err != nil {
			return nil, fmt.Errorf("error reading the CA Root file certificate: %v", err)
		}
		ok := certPool.AppendCertsFromPEM(rootCA)
		if !ok {
			return nil, fmt.Errorf("error appending the root CA to the certificate pool")
		}
		tlsConfig.RootCAs = certPool
	}
```

This reduces the code duplication and makes the intent clearer.